pkg/inventory/web/client.go

```diff
@@ -240,49 +241,46 @@ func (r *Client) Watch(
 	}
 	post := func(w *WatchReader) (pStatus int, pErr error) {
 		socket, response, pErr := dialer.Dial(url, header)
+		if response != nil {
+			pStatus = response.StatusCode
+			switch pStatus {
+			case http.StatusOK,
+				http.StatusSwitchingProtocols:
+				pStatus = http.StatusOK
+			default:
+				pErr = nil
+				return
+			}
+		}
```

Comment:
More accurate reporting.  Separate failures with routine HTTP results.

This fixes the following problem:
Use case:
- The watch is created and healthy.
- The web server is restarted and the connection is broken.
- The client attempts to re-establish the watch but the resource no longer exists resulting in a 404.  As a result, the client should abort.  Without this fix, the gets an `error` and retires indefinitely.